watertap/property_models/anaerobic_digestion/adm1_properties_vapor.py

```diff
@@ -0,0 +1,466 @@
+    """
+    StateBlock for calculating thermophysical properties associated with the ADM1
+    reaction system.
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super().build()
+
+        # Create state variables
+        self.flow_vol = pyo.Var(
+            initialize=1,
+            domain=pyo.NonNegativeReals,
+            doc="Total volumetric flowrate",
+            units=pyo.units.m**3 / pyo.units.s,
+        )
+        self.pressure = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=101325.0,
+            bounds=(1e3, 1e6),
+            doc="Pressure",
+            units=pyo.units.Pa,
+        )
+        self.temperature = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=298.15,
+            bounds=(298.15, 323.15),
+            doc="Temperature",
+            units=pyo.units.K,
+        )
+        self.conc_mass_comp = pyo.Var(
+            self.params.solute_set,
+            domain=pyo.NonNegativeReals,
+            initialize=0.1,
+            doc="Component mass concentrations",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+
+        self.p_w_sat = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=5643.8025,
+            doc="Water pressure",
+            units=pyo.units.Pa,
+        )
+
+        init = {"S_ch4": 65077, "S_co2": 36255, "S_h2": 1.639}
+
+        self.p_sat = pyo.Var(
+            self.params.solute_set,
+            domain=pyo.NonNegativeReals,
+            initialize=init,
+            doc="Component pressure",
+            units=pyo.units.Pa,
+        )
+
+        def p_sat_rule(b, j):
+            if j == "S_h2":
+                return self.p_sat[j] == pyo.units.convert(
+                    b.conc_mass_comp[j]
+                    * (1000 * pyo.units.g / pyo.units.kg)
+                    * Constants.gas_constant
+                    * b.temperature
+                    / (16 * pyo.units.g / pyo.units.mole),
+                    to_units=pyo.units.Pa,
+                )
+            elif j == "S_ch4":
+                return self.p_sat[j] == pyo.units.convert(
+                    b.conc_mass_comp[j]
+                    * (1000 * pyo.units.g / pyo.units.kg)
+                    * Constants.gas_constant
+                    * b.temperature
+                    / (64 * pyo.units.g / pyo.units.mole),
+                    to_units=pyo.units.Pa,
+                )
+            else:
+                return self.p_sat[j] == pyo.units.convert(
+                    b.conc_mass_comp[j]
+                    * (1000 * pyo.units.g / pyo.units.kg)
+                    * Constants.gas_constant
+                    * b.temperature
+                    / (12 * pyo.units.g / pyo.units.mole),
+                    to_units=pyo.units.Pa,
+                )
+
+        self._p_sat = pyo.Constraint(
+            self.params.solute_set, rule=p_sat_rule, doc="P for not solutes"
```

Comment:
Initially got confused by the indexing by `solute_set` and docstring of "P for not solutes"

Reply:
#918